designs/capi-resources-custom-namespace.md

```diff
@@ -0,0 +1,63 @@
+# CAPI Resources in Custom Namespace
+
+## Introduction
+
+**Problem:** Customers managing large scale EKS-A clusters (hundreds or thousands of workload clusters for a given management cluster) expect to have difficulty managing all these resources if they are all in the same namespace. Specifically, they expect to encounter two problems:
+
+1. Difficulty troubleshooting issues for a single cluster when there are so many resources to sift through. 
+2. High levels of load on a bootstrap cluster when executing an upgrade, since the clusterctl move operation will take all the CAPI components inside a namespace and instatiate them on the bootstrap cluster. This can result in heavy and unnecessary load on the bootstrap cluster
+
+This document aims to solve that, by allowing customers to instantiate the underlying CAPI/CAPX/etcd resources in a newly defined namespace, rather than all being dumped into `eksa-system`. 
+This change would make it so customers would be able to put these resources virtually in any namespace (except a few denylisted ones, discussed below), despite the fact that they may be considered eks-a internal components.
```

Comment:
@maxdrib I appreciate the PR title (+ parts of the discussion) suggests CAPX only but isn't this something that should be consistent among providers?

Reply:
I was using the term CAPX to capture all the different CAPI providers (CAPC, CAPV, etc.) - please let me know if there's a better one I could use instead

Reply:
I see. Perhaps drop CAPX (I keep thinking its a specific provider of some sort).